### PR TITLE
Project disable staff access option

### PR DESCRIPTION
**Pull request by bw2**
Fixes https://github.com/macarthur-lab/seqr/issues/390

### CODE REVIEW

File: xbrowse_server/base/models.py [hunk 1]
```diff
@@ -192,19 +194,19 @@ def can_view(self, user):
 
         if self.is_public:
             return True
-        elif user.is_staff:
+        elif (user.is_staff and not self.disable_staff_access) or ProjectCollaborator.objects.filter(project=self, user=user).exists():
             return True
         else:
             return ProjectCollaborator.objects.filter(project=self, user=user).exists()
```
Comment (hanars) [inline xbrowse_server/base/models.py]: you will check `ProjectCollaborators` here twice

File: xbrowse_server/base/models.py [hunk 2]
```diff
@@ -192,19 +194,19 @@ def can_view(self, user):
...
 
     def can_edit(self, user):
-        if user.is_staff:
+        if (user.is_staff and not self.disable_staff_access) or ProjectCollaborator.objects.filter(project=self, user=user).exists():
             return True
         else:
             return ProjectCollaborator.objects.filter(project=self, user=user).exists()
```
Comment (hanars) [inline xbrowse_server/base/models.py]: same here

File: xbrowse_server/base/models.py (hunk 1 above)
Comment (bw2) [reply xbrowse_server/base/models.py]: you're right.

File: xbrowse_server/base/models.py [hunk 3]
```diff
@@ -187,27 +190,45 @@ class Project(models.Model):
...
             return True
-        elif user.is_staff:
+
+        is_collaborator = self.is_collaborator(user)
+        if (user.is_staff and not self.disable_staff_access) or user.is_superuser or is_collaborator:
             return True
-        else:
-            return ProjectCollaborator.objects.filter(project=self, user=user).exists()
+
+        return is_collaborator
```
[the remark this answers is not in the document]
Comment (bw2) [reply xbrowse_server/base/models.py]: good call. Fixing..

File: seqr/views/utils/permissions_utils.py [hunk 4]
```diff
@@ -30,28 +31,38 @@ def get_project_and_check_permissions(project_guid, user, permission_level=CAN_V
...
     else:
-        return Project.objects.filter(can_edit_group__user=user)
+        return Project.objects.filter(can_edit_filter)
 
 
 def add_user_to_project(user, project, permission_level=CAN_VIEW):
     _validate_permissions_arg(permission_level)
 
-    if user.is_staff:
+    if user.is_superuser or user.is_staff:
```
[the remark this answers is not in the document]
Comment (hanars) [reply seqr/views/utils/permissions_utils.py]: so if you have a project with `disable_staff_access` how do you grant a normal staff memeber access?

[the remark this answers is not in the document]
Comment (bw2) [reply seqr/views/utils/permissions_utils.py]: oh, I see.. you're right. Thanks.

**Comment by bw2** [comment]
Fixed.

**Pull request merged**